isis/src/base/objs/SpicePosition/SpicePosition.cpp

```diff
@@ -359,25 +373,31 @@ namespace Isis {
     p_fullCacheSize = isdPos["SpkTableOriginalSize"].get<double>();
     p_cacheTime = isdPos["EphemerisTimes"].get<std::vector<double>>();
 
-    for (auto it = isdPos["Positions"].begin(); it != isdPos["Positions"].end(); it++) {
-      std::vector<double> pos = {it->at(0).get<double>(), it->at(1).get<double>(), it->at(2).get<double>()};
-      p_cache.push_back(pos);
-    }
-
-    p_cacheVelocity.clear();
+//    p_hasVelocity = !p_cacheVelocity.empty();
 
     bool hasVelocityKey = isdPos.find("Velocities") != isdPos.end();
+    p_hasVelocity = hasVelocityKey; 
```

Comment:
Can I make this assumption?

Reply:
yes. If you really want to be save, you can check that it's got a non-zero length.